e2etest/pkg/executor/executor.go

```diff
@@ -68,3 +68,25 @@ func (p *podExecutor) Exec(cmd string, args ...string) (string, error) {
 	out, err := exec.Command(Kubectl, fullargs...).CombinedOutput()
 	return string(out), err
 }
+
+// add ephemeral container to deal with distroless image
+type podDebugExecutor struct {
+	namespace string
+	name      string
+}
+
+func ForPodDebug(namespace, name string) Executor {
+	return &podDebugExecutor{
+		namespace: namespace,
+		name:      name,
+	}
+}
+
+func (pd *podDebugExecutor) Exec(cmd string, args ...string) (string, error) {
+	if Kubectl == "" {
+		return "", errors.New("the kubectl parameter is not set")
+	}
+	fullargs := append([]string{"debug", "-it", "-n", pd.namespace, pd.name, "-c", "pod-executor-debugger", "--image=busybox", "--", cmd}, args...)
```

Comment:
I think we should randomize the container name here `pod-executor-debugger-..`, the reason being that if you run the same debug container multiple times it only remembers the first output (the first debug command that caused the container to be terminated) which could cause confusion if the test is ran multiple times on the same cluster

Reply:
A randomized string (length=5) was added to the pod executor debugger.